### PR TITLE
Implemented physics collision visualizer

### PR DESCRIPTION
**Pull request by AndreaCatania**
Implemented the collision visualizer to better debug the collisions
ezgif com-video-to-gif 8
screenshot from 2019-01-14 09-55-36

### CODE REVIEW

**Comment by AndreaCatania** [comment]
I've added the Bullet Profiler.

I know that this is a new feature but consider that a collision visualizer and the profiler is a good feature to have in a stable release.

**Comment by AndreaCatania** [comment]
Added the AABB drawing 
screenshot from 2019-02-19 16-04-17

**Comment by AndreaCatania** [comment]
What's about this?

**Comment by AndreaCatania** [comment]
@akien-mga any chance for this PR to be merged if I resolve the conflict?

**Comment by akien-mga** [comment]
Sure.

**Comment by AndreaCatania** [comment]
Done

File: modules/bullet/SCsub [hunk 1]
```diff
@@ -187,8 +187,10 @@ if env['builtin_bullet']:
     thirdparty_sources = [thirdparty_dir + file for file in bullet2_src]
 
     env_bullet.Prepend(CPPPATH=[thirdparty_dir])
-    # if env['target'] == "debug" or env['target'] == "release_debug":
-    #     env_bullet.Append(CPPFLAGS=['-DBT_DEBUG'])
+
+    if env['target'] == "debug" or env['target'] == "release_debug":
+        #env_bullet.Append(CCFLAGS=['-DBT_DEBUG'])
+        env_bullet.Append(CCFLAGS=['-DBT_ENABLE_PROFILE'])
```
Comment (akien-mga) [inline modules/bullet/SCsub]: Those two should be `CPPFLAGS`.

**Comment by akien-mga** [comment]
You have two commits with the same name instead of one it seems.

File: modules/bullet/SCsub (hunk 1 above)
Comment (AndreaCatania) [reply modules/bullet/SCsub]: Done

**Comment by AndreaCatania** [comment]
It were two separate commits purposely since this introduce the collision visualizer and the profiler, but the commit message of the first commit got changes for an error. 

Now there is only one commit

**Comment by akien-mga** [comment]
There are a couple of warnings (errors on CI) about locals that shadow other locals: https://travis-ci.org/godotengine/godot/jobs/526779988

**Comment by akien-mga** [comment]
Code seems OK to me otherwise, though I think it would have been better to PR the collision visualizer and the profiler separately, as they don't seem directly related.

**Comment by AndreaCatania** [comment]
@akien-mga fixed last problems.
They are not related but I've already combined them so I would prefer if you merge them as is at this point.

File: scene/main/viewport.h [hunk 2]
```diff
@@ -186,6 +186,12 @@ class Viewport : public Node {
 	RID contact_2d_debug;
 	RID contact_3d_debug_multimesh;
 	RID contact_3d_debug_instance;
+	SpatialMaterial *contact_3d_debug_mat;
```
Comment (akien-mga) [inline scene/main/viewport.h]: This will likely fail compiling with `disable_3d=yes`, as `SpatialMaterial` is not registered in ClassDB then.

Comment (AndreaCatania) [reply scene/main/viewport.h]: I'm going to use _3D_DISABLED is it ok?

Comment (akien-mga) [reply scene/main/viewport.h]: Yeah that should do the trick.

Comment (AndreaCatania) [reply scene/main/viewport.h]: Done

File: modules/bullet/SCsub (hunk 1 above)
Comment (AndreaCatania) [reply modules/bullet/SCsub]: Also taken care of the cpp file

**Comment by AndreaCatania** [comment]
I'm tired to refactor this PR another time honestly

**Pull request closed without merging**